packages/core/admin/ee/admin/src/pages/SettingsPage/pages/SingleSignOn/SingleSignOnPage.tsx

```diff
@@ -0,0 +1,339 @@
+import { ProvidersOptions } from '../../../../../../../shared/contracts/admin';
+
+const schema = yup.object().shape({
+  autoRegister: yup.bool().required(translatedErrors.required),
+  defaultRole: yup.mixed().when('autoRegister', (value, initSchema) => {
+    return value ? initSchema.required(translatedErrors.required) : initSchema.nullable();
+  }),
+  ssoLockedRoles: yup
+    .array()
+    .nullable()
+    .of(
+      yup.mixed().when('ssoLockedRoles', (value, initSchema) => {
+        return value ? initSchema.required(translatedErrors.required) : initSchema.nullable();
+      })
+    ),
+});
+
+export const SingleSignOnPage = () => {
+  useFocusWhenNavigate();
+
+  const { formatMessage } = useIntl();
+  const permissions = useSelector(selectAdminPermissions);
+  const { lockApp, unlockApp } = useOverlayBlocker();
+  const toggleNotification = useNotification();
+
+  const { get, put } = useFetchClient();
+
+  const { isLoading: isLoadingProviderOptions, data } = useQuery(
+    ['providers', 'options'],
+    async () => {
+      const { data } = await get<ProvidersOptions.Response>('/admin/providers/options');
+
+      return data.data;
+    },
+    {
+      onError() {
+        toggleNotification({
+          type: 'warning',
+          message: { id: 'notification.error', defaultMessage: 'An error occured' },
+        });
+      },
+    }
+  );
+
+  const submitMutation = useMutation<
+    ProvidersOptions.Response['data'],
+    AxiosError<ProvidersOptions.Response>,
+    ProvidersOptions.Request['body']
+  >(
+    async (body) => {
+      const { autoRegister, defaultRole, ssoLockedRoles } = body;
+      const { data } = await put<ProvidersOptions.Response>('/admin/providers/options', {
+        autoRegister,
+        defaultRole,
+        ssoLockedRoles,
+      });
+
+      return data.data;
+    },
+    {
+      async onSettled() {
+        // @ts-expect-error – we're going to implement a context assertion to avoid this
+        unlockApp();
+      },
+    }
+  );
+
+  const {
+    isLoading: isLoadingPermissions,
+    allowedActions: { canUpdate, canReadRoles },
+  } = useRBAC({
+    ...permissions.settings?.sso,
+    readRoles: permissions.settings?.roles.read ?? [],
+  });
+
+  const { roles, isLoading: isLoadingRoles } = useAdminRoles(undefined, {
+    enabled: canReadRoles,
+  });
+
+  const handleSubmit = async (
+    body: ProvidersOptions.Request['body'],
+    { resetForm }: FormikHelpers<ProvidersOptions.Request['body']>
+  ) => {
+    // @ts-expect-error - context assertation
+    lockApp();
+
+    submitMutation.mutate(body, {
+      onSuccess(data) {
+        resetForm({ values: data });
+        toggleNotification({
+          type: 'success',
+          message: { id: 'notification.success.saved' },
+        });
+      },
+      onError() {
+        // const res = error?.response?.data;
+
+        // if (res?.data) {
+        //   return setErrors(res.data);
+        // }
```

Comment:
API validation errors need to be sent to the Formik bag. I couldn't find how to handle this with the existing codebase and I don't want to create a new handler if it already exists. 

https://strapihq.slack.com/archives/C042H3UGR1R/p1699616619907609

Reply:
You are right: validation is done by Formik before sending the mutation and this handler is only executed if the data matches the schema. In case it is hard to type just add a ts-ignore comment and move on.

Reply:
^^^ this

Reply:
I'm not sure I understand sorry. The client validation is done before sending the mutation but what about the API validation that could potentially trigger errors ?

Reply:
I mean, we need to handle API validation errors with Formik after the mutation is done. Is there already a way to handle that properly ?

Reply:
I haven't found this in the existing codebase, I have the impression that no api validation errors are properly handled, I hope I'm wrong.